src/rtctools_heat_network/workflows/utils/adapt_profiles.py

```diff
@@ -0,0 +1,155 @@
+
+logger = logging.getLogger("WarmingUP-MPC")
+logger.setLevel(logging.INFO)
+
+
+def set_data_with_averages_and_peak_day(
+    datastore: DataStore,
+    variable_name: str,
+    ensemble_member: int,
+    new_date_times: np.array,
+    problem: object,
+):
+    try:
+        data = problem.get_timeseries(variable=variable_name, ensemble_member=ensemble_member)
+    except KeyError:
+        datastore.set_timeseries(
+            variable=variable_name,
+            datetimes=new_date_times,
+            values=np.asarray([0.0] * len(new_date_times)),
+            ensemble_member=ensemble_member,
+            check_duplicates=True,
+        )
+        return
+
+    new_data = list()
+    data_timestamps = data.times
+    data_datetimes = [
+        problem.io.datetimes[0] + datetime.timedelta(seconds=s) for s in data_timestamps
+    ]
+    assert new_date_times[0] == data_datetimes[0]
+    data_values = data.values
+
+    values_for_mean = [0.0]
+    for dt, val in zip(data_datetimes, data_values):
+        if dt in new_date_times:
+            new_data.append(np.mean(values_for_mean))
+            values_for_mean = [val]
+        else:
+            values_for_mean.append(val)
+
+    # last datetime is not in input data, so we need to take the mean of the last bit
+    new_data.append(np.mean(values_for_mean))
+
+    datastore.set_timeseries(
+        variable=variable_name,
+        datetimes=new_date_times,
+        values=np.asarray(new_data),
+        ensemble_member=ensemble_member,
+        check_duplicates=True,
+    )
+
+
+def adapt_hourly_year_profile_to_day_averaged_with_hourly_peak_day(problem, problem_day_steps):
+    """
+    Adapt yearly porifle with hourly time steps to a common profile (daily averaged profile except
+    for the day with the peak demand).
+
+    Return the following:
+        - problem_indx_max_peak: index of the maximum of the peak values
+        - heat_demand_nominal: max demand value found for a specific heating demand
+    """
+
+    demands = problem.heat_network_components.get("demand", [])
+    new_datastore = DataStore(problem)
+    new_datastore.reference_datetime = problem.io.datetimes[0]
+
+    for ensemble_member in range(problem.ensemble_size):
+        problem_indx_max_peak = -1.0
+        day_steps = -1.0
+        parameters = problem.parameters(ensemble_member)
+        total_demand = None
+        heat_demand_nominal = dict()
+
+        for demand in demands:
+            try:
+                demand_values = problem.get_timeseries(
+                    f"{demand}.target_heat_demand", ensemble_member
+                ).values
+            except KeyError:
+                continue
+            if total_demand is None:
+                total_demand = demand_values
+            else:
+                total_demand += demand_values
+            heat_demand_nominal[f"{demand}.Heat_demand"] = max(demand_values)
+            heat_demand_nominal[f"{demand}.Heat_flow"] = max(demand_values)
```

Comment:
@jimrojerTNO: this line has been added to be the same as master, which originally comes from the branch "Bug fix inconsistent ub (#36)"